errors.go

```diff
@@ -80,3 +80,12 @@ type ErrManifestBlobUnknown struct {
 func (err ErrManifestBlobUnknown) Error() string {
 	return fmt.Sprintf("unknown blob %v on manifest", err.Digest)
 }
+
+// ErrOptionFailedToApply is returned when a function option returns an error
+type ErrOptionFailedToApply struct {
```

Comment:
This seems unnecessary. What action would a caller take when encountering this error?